csharp/ql/src/experimental/dataflow/LocalCommandInjectionQuery.qll

```diff
@@ -0,0 +1,81 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about local command injection vulnerabilities.
+ */
+
+import csharp
+private import semmle.code.csharp.security.dataflow.flowsources.Local
+private import semmle.code.csharp.frameworks.system.Diagnostics
+private import semmle.code.csharp.security.Sanitizers
+
+/**
+ * A source specific to command injection vulnerabilities.
+ */
+abstract class Source extends DataFlow::Node { }
+
+/**
+ * A sink for command injection vulnerabilities.
+ */
+abstract class Sink extends DataFlow::ExprNode { }
+
+/**
+ * A sanitizer for user input treated as code vulnerabilities.
+ */
+abstract class Sanitizer extends DataFlow::ExprNode { }
+
+/**
+ * A taint-tracking configuration for command injection vulnerabilities.
+ */
+module LocalCommandInjectionConfig implements DataFlow::ConfigSig {
+  /**
+   * Holds if `source` is a relevant data flow source.
+   */
+  predicate isSource(DataFlow::Node source) { source instanceof Source }
+
+  /**
+   * Holds if `sink` is a relevant data flow sink.
+   */
+  predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
+
+  /**
+   * Holds if data flow through `node` is prohibited. This completely removes
+   * `node` from the data flow graph.
+   */
+  predicate isBarrier(DataFlow::Node node) { node instanceof Sanitizer }
+}
+
+/**
+ * A taint-tracking module for local command injection vulnerabilities.
+ */
+module LocalCommandInjection = TaintTracking::Global<LocalCommandInjectionConfig>;
+
+/** A source of local user input. */
+class LocalSource extends Source instanceof LocalFlowSource { }
+
+/**
```

Comment:
Looks like exactly as https://github.com/github/codeql/blob/d90ddf1b431dacf33b464453a18845b04e50d5fb/csharp/ql/lib/semmle/code/csharp/security/dataflow/CommandInjectionQuery.qll#L75-L98
Is it private and cannot be imported? Just wonder why this copy paste...

Reply:
Because I did use the existing Remote.qll as the template, this is my first time modifying libraries (Not encapsulating everything in a standalone query), and I thought it would be more useful this way. For example, to detect path traversals with local sources as well. 

Should I get rid of that part?

Reply:
I'll let CodeQL team to comment what is the preferred way. I think importing already existing class is preferrable unless you had to modify some methods of them. In that case CodeQL team, I think, would like to keep everything in the experimental folder.
From my perspective it is harder to understand what is added in the query. Since we had local taint sources and the `ProcessStartInfo` was already modeled, I'm trying to understand what is new.

Reply:
The method for reading system arguments from the Main class :)